### ISSUE TITLE
Go backend naming inconsistency (in `mise ls` and `mise prune`)

### ISSUE THREAD

**Issue by dhanak** [jdx/mise#1904]
**Describe the bug**
I have some packages installed with the go backend, e.g., `go:github.com/tomnomnom/gron`. It is listed as `go:github.com/tomnomnom/gron = 'latest'` in my `mise/config.toml`, it installs as expected and works fine. However, when I list the installed packages, I see this:

```
go:github.com-tomnomnom-gron       latest                                                                      
go:github.com/tomnomnom/gron       latest         ~/.config/mise/config.toml                      latest
```

In other words, it shows up as I entered it, but also with dashes instead of slashes, and the latter is shown as missing. Furthermore, if I run `mise prune`, it asks me if I want to remove the dashed "variant". Answering yes removes the `gron` installation, and now the slashed version shows up as missing. Running `mise install` again restarts the cycle.

**To Reproduce**

```
$ mise use go:github.com/tomnomnom/gron
$ mise ls
$ mise prune
```

**Expected behavior**

Only `go:github.com/tomnomnom/gron` is shown in the listing.

**`mise doctor` output**

```text
version: 2024.4.4 linux-x64 (37b44e2 2024-04-11)
activated: yes
shims_on_path: no

build_info: 
  Target: x86_64-unknown-linux-gnu
  Features: DEFAULT, NATIVE_TLS, OPENSSL
  Built: Thu, 11 Apr 2024 17:14:58 +0000
  Rust Version: rustc 1.77.2 (25ef9e3d8 2024-04-09)
  Profile: release

shell: 
  /bin/zsh
  zsh 5.9 (x86_64-debian-linux-gnu)

dirs: 
  data: ~/.local/share/mise
  config: ~/.config/mise
  cache: ~/.cache/mise
  state: ~/.local/state/mise
  shims: ~/.local/share/mise/shims

config_files: 
  ~/.config/mise/config.toml

plugins: 
  bun                                (core)
  cargo:mdcat                        (core)
  cargo:ouch                         (core)
  cmake                              https://github.com/srivathsanmurali/asdf-cmake.git#3811ad7
  deno                               (core)
  elixir                             https://github.com/asdf-vm/asdf-elixir.git#a4c42e1
  erlang                             (core)
  eza                                https://github.com/lwiechec/asdf-eza.git#dedb2d1
  fd                                 https://gitlab.com/wt0f/asdf-fd.git#def644b
  fzf                                https://github.com/kompiro/asdf-fzf.git#d19eb67
  go                                 (core)
  go:github.com-brimdata-zed-cmd-zq  (core)
  go:github.com-tomnomnom-gron       (core)
  java                               (core)
  julia                              https://github.com/rkyleg/asdf-julia.git#97cf19a
  just                               https://github.com/olofvndrhr/asdf-just.git#93771e1
  node                               (core)
  python                             (core)
  rebar                              https://github.com/Stratus3D/asdf-rebar.git#4af0770
  ripgrep                            https://gitlab.com/wt0f/asdf-ripgrep.git#e836665
  ruby                               (core)
  starship                           https://github.com/grimoh/asdf-starship.git#9ffad1c
  usage                              https://github.com/jdx/mise-usage.git#fe3888a
  vale                               https://github.com/pdemagny/asdf-vale#9926c81
  vhs                                https://github.com/chessmango/asdf-vhs.git#8c1d33e

toolset: 
  julia@1.7.3           
  cargo:mdcat@2.1.1     
  cargo:ouch@0.5.1      
  go:github.com/brimdata/zed/cmd/zq@latest  
  go:github.com/tomnomnom/gron@latest  
  cmake@3.24.1          
  elixir@1.16.0-otp-25  
  erlang@25.0           
  erlang@23.2.1         
  eza@0.18.10           
  fd@9.0.0              
  fzf@0.50.0            
  go@1.22.2             
  java@22.0.0           
  node@21.7.3           
  python@3.11.4         
  rebar@3.20.0          
  rebar@3.16.1          
  ripgrep@14.1.0        
  ruby@3.3.0            
  starship@1.18.2       
  usage@0.1.18          
  vale@3.4.1            
  vhs@0.3.0             

env_vars: 
  MISE_SHELL=zsh

settings: 
  activate_aggressive = false
  all_compile = false
  always_keep_download = false
  always_keep_install = false
  asdf_compat = false
  cargo_binstall = true
  color = true
  disable_default_shorthands = false
  disable_tools = []
  experimental = true
  go_default_packages_file = "~/.default-go-packages"
  go_download_mirror = "https://dl.google.com/go"
  go_repo = "https://github.com/golang/go"
  go_set_gopath = false
  go_set_goroot = true
  go_skip_checksum = false
  jobs = 4
  legacy_version_file = true
  legacy_version_file_disable_tools = []
  node_compile = false
  not_found_auto_install = false
  paranoid = false
  plugin_autoupdate_last_check_duration = "7d"
  python_compile = false
  python_default_packages_file = "/home/david/.default-python-packages"
  python_pyenv_repo = "https://github.com/pyenv/pyenv.git"
  raw = false
  trusted_config_paths = ["~/CI"]
  quiet = false
  verbose = false
  yes = false
  ci = false
  debug = false
  trace = false
  log_level = "info"
  python_venv_auto_create = false

  [status]
  missing_tools = "if_other_versions_installed"
  show_env = false
  show_tools = false

No problems found
```

**Additional context**

**Comment by roele** [comment]
This clearly came in with #1885 where installed backend binaries are listed by default. 

This is done by reading them from the `$MISE_DATA_DIR/installs` directory. While installing, the backend package name is turned into a folder name by replacing `:` and `/` with `-`. Now for `Go`, package names cannot contain a `-` and simply doing the reverse will fix the issue. But i'm afraid that we still have an issue with other backends like NPM which contain both `/` and `-` (e.g. `npm:@adobe/aio-cli` --> `npm-@adobe-aio-cli`).

**Comment by dhanak** [comment]
I confirm this is fixed in 2024.4.5.
